Update FAQs, Remove transitional instructions

milljm opened the pull request:
Made the FAQs, more FAQ-ish and easier on the eyes.

Removed the transition away from moose-environment package. It was there
long enough.

Added a new section about us switching out the Conda channel name, and
how to fix possible errors.

Closes #15929

Commit:
Upate FAQs, Remove transitional instructions

Made the FAQs, more FAQ-ish and easier on the eyes.

Removed the transition away from moose-environment package. It was there
long enough.

Added a new section about us switching out the Conda channel name, and
how to fix possible errors.

Closes #15929

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/593929/) on 024aaac wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15931/site)

This comment will be updated on new commits.

cticenhour commented on modules/doc/content/getting_started/installation/conda.md:
```diff
@@ -1,19 +1,12 @@
 # Conda MOOSE Environment
 
-Our preferred method for obtaining libraries necessary for MOOSE based
+Our preferred method for obtaining dependencies necessary for MOOSE based
```
```suggestion
Our preferred method for obtaining dependencies necessary for MOOSE-based
```

cticenhour reviewed (changes requested):
So much more FAQ-ey! 😄

Commit:
Spelling/Grammar corrections and additions

Addressed comments about spelling and other issues.
Added link from FAQ Conda, to Getting Started instructions.

Refs #15929

cticenhour reviewed (approved):
Looks good to me!
